src/main/java/touro/peg/BoardSolutionChecker.java

```diff
@@ -0,0 +1,84 @@
+package touro.peg;
+
+import touro.peg.TriangleBoard;
+import touro.tree.TriangleBoardTree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BoardSolutionChecker
+{
+    private TriangleBoardTree tree;
+    public BoardSolutionChecker(TriangleBoard board)
+    {
+        this.tree = new TriangleBoardTree(board);
+    }
+
+    public List<TriangleBoard> getSolutions()
+    {
+        ArrayList<TriangleBoard> solutions = new ArrayList<>();
+        if (tree == null)
+        {
+            return solutions;
+        }
+        List<TriangleBoardTree.TriangleTreeNode> leaves = tree.getLeaves();
+        for (TriangleBoardTree.TriangleTreeNode node : leaves)
+        {
+            if (node.getTriangleBoard().isWin() && !contains(solutions, node.getTriangleBoard()))
+            {
+                solutions.add(node.getTriangleBoard());
+            }
+        }
+        return solutions;
+    }
+
+    public List<TriangleBoard> getBestSolutions()
+    {
+        ArrayList<TriangleBoard> bestSolutions = new ArrayList<>();
+        if (tree == null)
+        {
+            return bestSolutions;
+        }
+        List<TriangleBoardTree.TriangleTreeNode> leaves = tree.getLeaves();
+        for (TriangleBoardTree.TriangleTreeNode node : leaves)
+        {
+            if (node.getTriangleBoard().isBestWin())
+            {
+                bestSolutions.add(node.getTriangleBoard());
+            }
+        }
+        return bestSolutions;
+    }
+
+    public ArrayList<List<TriangleBoardTree.TriangleTreeNode>> getPathsToBestSolutions()
+    {
+        ArrayList<List<TriangleBoardTree.TriangleTreeNode>> paths = new ArrayList<>();
+        if (tree == null)
+        {
+            return paths;
+        }
+        List<TriangleBoardTree.TriangleTreeNode> leaves = tree.getLeaves();
+        for (TriangleBoardTree.TriangleTreeNode node : leaves)
+        {
+            if (node.getTriangleBoard().isBestWin())
+            {
+                paths.add(tree.getSequenceToNode(node));
+            }
+        }
+        return paths;
+    }
+
+    public boolean contains(List<TriangleBoard> boards, TriangleBoard board)
```

Comment:
List has a contains method already

Reply:
Yes, but I was struggling to get it to work properly so I wrote my own temporary one.
I think someone overrided the .equals() method with the equalsBoard method so maybe that would help?